utils/testgen.py

```diff
@@ -237,6 +246,34 @@ def pytest_generate_tests(metafunc):
             # Skip unwanted types
             continue
 
+        # Test to see the test has meta data, if it does and that metadata contains
+        # a test_flag kwarg, then check to make sure the provider contains that test_flag
+        # if not, do not collect the provider for this particular test.
+
+        # Obtain the tests flags
+        meta = getattr(metafunc.function, 'meta', None)
+
+        test_flags = getattr(meta, 'kwargs', {}) \
+            .get('from_docs', {}).get('test_flag', '').split(',')
+        test_flags = [flag.strip() for flag in test_flags]
+
+        defined_flags = cfme_data.get('test_flags', '').split(',')
+        defined_flags = [flag.strip() for flag in defined_flags]
+
+        excluded_flags = data.get('excluded_test_flags', '').split(',')
+        excluded_flags = [flag.strip() for flag in excluded_flags]
+
+        allowed_flags = set(defined_flags) - set(excluded_flags)
+
+        if set(test_flags) - allowed_flags:
+            logger.info("Skipping Provider {} for test {} because "
```

Comment:
I think this should maybe be debug, and it should definitely include the module that contains the function.